utils/s2n_mem.c

```diff
@@ -75,7 +75,7 @@ static int s2n_mem_free_mlock_impl(void *ptr, uint32_t size)
 {
     int munlock_rc = munlock(ptr, size);
     free(ptr);
-    POSIX_GUARD(munlock_rc);
+    POSIX_ENSURE(munlock_rc >= S2N_SUCCESS, S2N_ERR_MUNLOCK);
```

Comment:
I wouldn't actually use S2N_SUCCESS here, since munlock is a posix method. You should probably just use 0, and check that it's == 0 (the man page doesn't specify what positive values would mean).

Reply:
Thanks, fixed as suggested.